scripts/ps_audit_log_v2.sh

```diff
@@ -0,0 +1,167 @@
+                ["audit_log_filter_syslog_tag"]="audit-filter")
+
+functions_list=("audit_log_rotate" "audit_log_encryption_password_set" \
+    "audit_log_encryption_password_get" "audit_log_filter_remove_user" \
+    "audit_log_read_bookmark" "audit_log_filter_set_user" \
+    "audit_log_filter_set_filter" "audit_log_filter_remove_filter" \
+    "audit_log_filter_flush" "audit_log_read")
+
+# Chech that all variables are present and have correct default values
+for al_variable in ${!variables[@]}; do
+    result=$(mysql -uroot -NBe "SELECT @@${al_variable}")
+    if [[ "${result}" != ${variables[${al_variable}]} ]]; then
+        echo "${al_variable} result is incorrect. Current result is: ${result}. Expected result is:  ${variables[${al_variable}]}}"
+        fails='1'
+    fi
+done
+
+# Chech that all UDFs are present
+for al_function in ${functions_list[@]}; do
+    result=$(mysql -uroot -NBe "SELECT * FROM performance_schema.user_defined_functions WHERE udf_name=\"${al_function}\";")
+    if [[ -z ${result} ]]; then
+        fails='1'
+        echo "${al_function} has empty result."
+    fi
+done
+
+# Functional tests
+
+# get logfile location
+data_dir=$(mysql -uroot -NBe "SELECT @@datadir";)
+al_file_name=$(mysql -uroot -NBe "SELECT @@audit_log_filter_file;")
+al_file_pattern=$(echo ${al_file_name}|awk -F'.' '{print $1}')
+al_file_location=${data_dir}${al_file_name}
+
+# Enable audit_log_v2 for all users and check that log file grows.
+al_file_rows=$(wc -l ${al_file_location}|awk '{print $1}')
+mysql -uroot -NBe "SELECT audit_log_filter_set_filter('log_all', '{ \"filter\": { \"log\": true } }');"
+mysql -uroot -NBe "SELECT audit_log_filter_set_user('%', 'log_all');"
+mysql -uroot -NBe "SELECT 1;"
+sleep 1
+al_file_rows_after=$(wc -l ${al_file_location}|awk '{print $1}')
+if [[ ${al_file_rows} -ge ${al_file_rows_after} ]]; then
+    echo "Enabled plugin. Smth wrong with audit log file growth! Rows num before SQL=${al_file_rows}. Rows num after SQL=${al_file_rows_after}!"
+    exit 1
+else
+    echo "Enabled plugin OK. Rows num before SQL=${al_file_rows}. Rows num after SQL=${al_file_rows_after}"
+fi
+
+# Check dynamic variables
+
+# audit_log_filter_disable
+mysql -uroot -NBe "set global audit_log_filter_disable='ON';"
+sleep 1
+al_file_rows=$(wc -l ${al_file_location}|awk '{print $1}')
+mysql -uroot -NBe "SELECT 1;"
+sleep 1
+al_file_rows_after=$(wc -l ${al_file_location}|awk '{print $1}')
+if [[ ${al_file_rows_after} -gt ${al_file_rows} ]]; then
+    echo "audit_log_filter_disable. Smth wrong with audit log file growth! Rows num before SQL=${al_file_rows}. Rows num after SQL=${al_file_rows_after}!"
+    exit 1
+else
+    echo "audit_log_filter_disable OK. Rows num before SQL=${al_file_rows}. Rows num after SQL=${al_file_rows_after}"
+fi
+# revert setting and enable audit log
+mysql -uroot -NBe "set global audit_log_filter_disable='OFF';"
```

Comment:
Maybe instead use: mysql -uroot -NBe "set global audit_log_filter_disable='default';"
We dont know what the default value maybe in the future

Reply:
The idea here is to definitely enable the audit logging for the future tests, otherwise they will all fail as logging will be disabled....

Reply:
Discussed in DM. Agreed to leave as is as defaults are checked earlier in script.